includes/save.php

```diff
@@ -337,116 +277,63 @@ function tsml_save_post($post_id, $post, $update) {
+				foreach ($old_meeting->location_meetings as $meeting) update_post_meta($meeting['id'], 'group_id', $contact_entity_id); 	
 			}
 		}
 
-		//group website 2
-		if (!$update || strcmp($old_meeting->website_2, $_POST['website_2']) !== 0) {
-			$changes[] = 'website_2';
-			if (empty($_POST['website_2'])) {
-				delete_post_meta($group_id, 'website_2');
-			} else {
-				update_post_meta($group_id, 'website_2', esc_url_raw($_POST['website_2'], array('http', 'https')));
-			}
-		}
-		
-		//group email
-		if (!$update || strcmp($old_meeting->email, $_POST['email']) !== 0) {
-			$changes[] = 'email';
-			if (empty($_POST['email'])) {
-				delete_post_meta($group_id, 'email');
-			} else {
-				update_post_meta($group_id, 'email', sanitize_text_field($_POST['email']));
+		//switching from individual meeting
+		if ($update && empty($old_meeting->group)) {
+			foreach ($tsml_contact_fields as $field => $type) {
+				//clear out contact information associated with meeting
+				delete_post_meta($post->ID, $field);
 			}
 		}
-		
-		//group phone
-		if (!$update || strcmp($old_meeting->phone, $_POST['phone']) !== 0) {
-			$changes[] = 'phone';
-			if (empty($_POST['phone'])) {
-				delete_post_meta($group_id, 'phone');
-			} else {
-				update_post_meta($group_id, 'phone', sanitize_text_field($_POST['phone']));
-			}
-		}
-		
-		//group mailing address
-		if (!$update || strcmp($old_meeting->mailing_address, $_POST['mailing_address']) !== 0) {
-			$changes[] = 'mailing_address';
-			if (empty($_POST['mailing_address'])) {
-				delete_post_meta($group_id, 'mailing_address');
-			} else {
-				update_post_meta($group_id, 'mailing_address', sanitize_text_field($_POST['mailing_address']));
-			}
-		}
-		
-		//group venmo
-		if (!$update || strcmp($old_meeting->venmo, $_POST['venmo']) !== 0) {
-			$changes[] = 'venmo';
-			if (empty($_POST['venmo']) || (substr($_POST['venmo'], 0, 1) != '@')) {
-				delete_post_meta($group_id, 'venmo');
-			} else {
-				update_post_meta($group_id, 'venmo', sanitize_text_field($_POST['venmo']));
-			}
-		}
-		
-		//contact info
-		for ($i = 1; $i <= GROUP_CONTACT_COUNT; $i++) {
-			foreach (array('name', 'email', 'phone') as $field) {
-				$key = 'contact_' . $i . '_' . $field;
-				$_POST[$key] = sanitize_text_field($_POST[$key]);
-				if (!$update || $old_meeting->{$key} != $_POST[$key]) {
-					$changes[] = $key;
-					if (empty($_POST[$key])) {
-						delete_post_meta($group_id, $key); 
-					} else {
-						update_post_meta($group_id, $key, $_POST[$key]);
-					}
-				}
-			}
+	}
+
+	//special validation, todo warn user on fail
+	if (!empty($_POST['venmo']) && substr($_POST['venmo'], 0, 1) != '@') {
+		$_POST['venmo'] = null;
+	}
+	if (!empty($_POST['square']) && substr($_POST['square'], 0, 1) != '$') {
+		$_POST['square'] = null;
+	}
+	if (!empty($_POST['paypal']) && strpos($_POST['paypal'], '/') !== false) {
+		$_POST['paypal'] = array_pop(explode('/', $_POST['paypal']));
+	}
```

Comment:
If we're switching this field to just be the paypal.me username, and not the full URL, we'll have to change this. This is still planning on a full URL.

Reply:
What this line does is it says if `paypal` contains a `/` then take everything after the last occurrence. The intent was if you entered `https://www.paypal.com/foobar` that it would grab the value `foobar`